src/particles/spacecharge/Gauss2p5dPush.cpp

```diff
@@ -0,0 +1,263 @@
+
+        amrex::ParticleReal fy = 0.0;
+        amrex::ParticleReal fz = 0.0;
+        amrex::ParticleReal t2sqrt = 0.0;
+        amrex::ParticleReal f2y = 0.0;
+        amrex::ParticleReal asp2 = asp*asp;
+        amrex::ParticleReal asp2m = 1_prt - asp*asp;
+        for (int i = 0; i < nint; ++i)
+        {
+            x = xmin + i*h;
+            t2 = asp2 + asp2m * x*x;
+            t2sqrt = std::sqrt(t2);
+            f2y = t2*t2sqrt;
+            f2 = x*t2sqrt;
+            f1 = std::exp(-x*x*(xp2 + yp2 / t2) / 2_prt);
+            f1x = x*f1;
+
+            // Ez
+            fz = (f1 - 1_prt) / f2;
+            if (i%2 == 0)
+              sum0 += 2_prt * fz * h;
+            else
+              sum0 += 4_prt * fz * h;
+
+            // Ex
+            fx = f1x / t2sqrt;
+            if (i%2 == 0)
+              sum0ex += 2_prt * fx * h;
+            else
+              sum0ex += 4_prt * fx * h;
+
+            // Ey
+            fy = f1x / f2y;
+            if (i%2 == 0)
+              sum0ey += 2_prt * fy * h;
+            else
+              sum0ey += 4_prt * fy * h;
+        }
+
+        pintez = 2 * asp * sum0 / 3_prt;
+        pintex = 2 * asp * xp * sum0ex / 3_prt / sigx;
+        pintey = 2 * asp * yp * sum0ey / 3_prt / sigy;
+    }
+
+    //advanced particle momentae using 2.5D SC fields with a transverse Gaussian distribution.
+    void Gauss2p5dPush (ImpactXParticleContainer & pc, amrex::ParticleReal const slice_ds)
+    {
+        BL_PROFILE("impactx::spacecharge::Gauss2p5dPush");
+
+        using namespace amrex::literals;
+
+        amrex::ParticleReal const charge = pc.GetRefParticle().charge;
+
+        std::unordered_map<std::string, amrex::ParticleReal> const rbc = diagnostics::reduced_beam_characteristics(pc);
+        // get RMS sigma of beam
+        // Standard deviation of the particle positions (speed of light times time delay for t) (unit: meter)
+        amrex::ParticleReal const sigx = rbc.at("sig_x");
+        amrex::ParticleReal const sigy = rbc.at("sig_y");
+
+        // physical constants and reference quantities
+        amrex::ParticleReal const c0_SI = 2.99792458e8;  // TODO move out
+        amrex::ParticleReal const mc_SI = pc.GetRefParticle().mass * c0_SI;
+        amrex::ParticleReal const pz_ref_SI = pc.GetRefParticle().beta_gamma() * mc_SI;
+        amrex::ParticleReal const gamma = pc.GetRefParticle().gamma();
+        amrex::ParticleReal const inv_gamma2 = 1.0_prt / (gamma * gamma);
+        amrex::ParticleReal const rfpiepslon = c0_SI*c0_SI*1.0e-7;
+
+        amrex::ParticleReal const dt = slice_ds / pc.GetRefParticle().beta() / c0_SI;
+
+        int nint = 101;
+        amrex::Real delta = 0.01;
+        amrex::ParmParse pp_algo("algo.space_charge");
+        pp_algo.queryAddWithParser("gauss2p5_nint", nint);
+        pp_algo.queryAddWithParser("gauss2p5_delta", delta);
+
+        int tp5d_bins = 129;
+        pp_algo.queryAddWithParser("gauss2p5_bins", tp5d_bins);
```

Comment:
Discussed today as possible namings:
```suggestion
        pp_algo.queryAddWithParser("gauss_nint", nint);
        pp_algo.queryAddWithParser("gauss_taylor_delta", delta);

        int tp5d_bins = 129;
        pp_algo.queryAddWithParser("gauss_charge_z_bins", tp5d_bins);
```
- `gauss_nint`: can reuse in 3D and 2.5D
- `gauss_taylor_delta`: to avoid confusion -- we do not mean the integral range / gauss_nint
- `gauss_charge_z_bins`: more explicit